nio_cli/utils/ssl.py

```diff
@@ -5,52 +5,109 @@
 from cryptography.hazmat.primitives.asymmetric import rsa
 from datetime import datetime, timedelta
 from ipaddress import ip_address
-from os import makedirs, getcwd
-from os.path import join
+from os.path import join, isfile, dirname
 from platform import system
 from subprocess import run
 from .inputs import get_boolean, get_string
+from .files import ensure_dir_exists, DEFAULT_ROOT
 
 
 def config_ssl(root):
-    makedirs(join(root, 'etc/ssl'), exist_ok=True)
+    # See if they want to specify a custom CA to sign with
+    ca_cert_path = get_string(
+        "Enter the path to your CA certificate. If the CA does not exist "
+        "one will be created there for you",
+        default=join(DEFAULT_ROOT, 'ca.crt'))
+    ca_key_path = get_string(
+        "Enter the path to your CA private key. If the key does not exist "
+        "one will be created there for you",
+        default=join(DEFAULT_ROOT, 'ca.key'))
+    if not isfile(ca_cert_path):
+        print("No CA certificate found, creating one...")
+        ca_crt, ca_key = _create_ca()
+        ensure_dir_exists(dirname(ca_cert_path))
+        ensure_dir_exists(dirname(ca_key_path))
+        _save_cert(ca_cert_path, ca_crt)
+        _save_key(ca_key_path, ca_key)
+        _trust_cert(ca_cert_path)
+    else:
+        ca_crt = _read_cert(ca_cert_path)
+        ca_key = _read_key(ca_key_path)
+
     host = get_string(
         "Enter the host where you will access your instance",
         default="localhost")
-    cert_crt, cert_key = _create_cert(host)
-    cert_path = 'etc/ssl/cert.crt'
-    key_path = 'etc/ssl/cert.key'
-    _save_key(root, key_path, cert_key)
-    _save_cert(root, cert_path, cert_crt)
 
+    cert_crt, cert_key = _create_instance_cert(host, ca_crt, ca_key)
+    cert_path = join(root, 'etc', 'ssl', 'cert.crt')
```

Comment:
It looks like `root` isn't passing the full path correctly. On `nio new myInstance --ssl` the ssl paths show `myInstance/etc/ssl/cert.*` and on `nio config --ssl` I get `./etc/ssl/cert*` but it's missing the value of my `~` pretended to it.

Reply:
However this might be caused by #121

Reply:
I'm getting a `FileNotFoundError` when it attempts to load the cert when referenced relatively. But If I adjust it to the full path, nio starts up correctly

Reply:
Ahh okay so I have the issue. The nio new path with the cert paths set to `myInstanceName/etc/ssl/...` cause a FileNotFoundError but when you do nio config and the path is set to `./etc/ssl/...` everything starts up okay. I think we just need to keep that path using `./` rather than the name to begin the path

Reply:
better yet, we can just change the path generation to:
```python
cert_path = join('etc', 'ssl', 'cert.crt')
```

Reply:
same issue for me